code/modules/mob/say.dm

```diff
@@ -94,6 +94,9 @@
 		if(istype(other, src.type) || istype(src, other.type))
 			return TRUE
 		return FALSE
+	else	//AI and pAI check
+		if(isAI(src) || ispAI(src))
```

Comment:
Isn't that better to set `universal_speak` or `universal_understand` to `TRUE` for them?
Also what about pAI `universal translator` program?

Reply:
Oh, right! Fixed